lms/services/course.py

```diff
@@ -208,9 +208,11 @@ def get_by_id(self, id_: int) -> Course | None:
 
         return None
 
-    def is_member(self, course: Course, user: User) -> bool:
-        """Check if a user is a member of a course."""
-        return bool(course.memberships.filter_by(user=user).first())
+    def is_member(self, course: Course, h_userid: str) -> bool:
+        """Check if an H user is a member of a course."""
+        return bool(
+            course.memberships.join(User).filter(User.h_userid == h_userid).first()
```

Comment:
This is the actual behaviour change, we were returning a list of courses filtered by h_userid but checking membership with user (user.id).

Use h_userid in both with this change.